src/15utility.js

```diff
@@ -1082,6 +1082,13 @@ var extend = (utils.extend = function extend(a, b) {
 	return a;
 });
 
+/**
+ * Extracts the primitive data
+ */
+var getValueOf = (utils.getValueOf = function (val) {
```

Comment:
How come we need to have a new way of accessing the value?

Reply:
String and Number objects can't be compared the same way as primitives. So we need to convert them to primitives with valueOf() before comparing.